### PR TITLE
Read aperture size in serialisable format

### PR DESCRIPTION
**Pull request by dperl-dls**
Needed for [Hyperion 1311](https://github.com/DiamondLightSource/hyperion/issues/1311)
- Add a Converter and method to convert SingleAperturePosition
- Add a util to get beam size x and y from the above

### Instructions to reviewer on how to test:
1. Run tests

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly

### CODE REVIEW

File: src/dodal/beamline_specific_utils/i03.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+@dataclass
+class BeamSize:
+    x_um: float | None
+    y_um: float | None
+
+
+def beam_size_from_aperture(position: SingleAperturePosition):
+    aperture_size = position.radius_microns
+    return BeamSize(aperture_size, I03_BEAM_HEIGHT if aperture_size else None)
```
Comment (olliesilvester) [inline src/dodal/beamline_specific_utils/i03.py:16]: I don't think this file has to be an i03-specific util. We could read the `I03_BEAM_HEIGHT` from some parameter and then have this as a generic util

File: src/dodal/devices/aperturescatterguard.py [hunk 2]
```diff
@@ -93,18 +101,30 @@ def __init__(self, prefix: str = "", name: str = "") -> None:
...
-        self.selected_aperture = self.SelectedAperture(
-            backend=SoftSignalBackend(
-                SingleAperturePosition, AperturePositions.UNKNOWN
-            ),
+        aperture_backend = SoftSignalBackend(
+            SingleAperturePosition, AperturePositions.UNKNOWN
         )
+        aperture_backend.converter = self.ApertureConverter()
+        self.selected_aperture = self.SelectedAperture(backend=aperture_backend)
+        self.selected_aperture
```
Comment (olliesilvester) [inline src/dodal/devices/aperturescatterguard.py:109]: Remove typo

File: src/dodal/devices/aperturescatterguard.py [hunk 3]
```diff
@@ -38,6 +38,14 @@ class SingleAperturePosition:
         0, 0, 0, 0, 0
     )
 
+    def dict(self):
```
Comment (olliesilvester) [inline src/dodal/devices/aperturescatterguard.py:41]: Can we use the `dataclass.asdict()` method for this instead?

**Review by olliesilvester — APPROVED** [review]
Looks good, thanks. There's one typo to remove and an optional comment about using `asdict`

File: src/dodal/beamline_specific_utils/i03.py (hunk 1 above)
Comment (dperl-dls) [reply src/dodal/beamline_specific_utils/i03.py:16]: yeah, I kind of agree, there will be more beamlines that could use this, but I think it's easy enough to move it when we need it

File: src/dodal/devices/aperturescatterguard.py (hunk 3 above)
Comment (dperl-dls) [reply src/dodal/devices/aperturescatterguard.py:41]: yeah, I had actually already done that, just forgot to push :man_facepalming:

**Pull request merged**